hpy/devel/src/runtime/ctx_type.c

```diff
@@ -146,6 +165,26 @@ static inline HPyType_BuiltinShape _HPyType_Get_Shape(PyTypeObject *tp) {
     return _is_HPyType(tp) ? _HPyType_EXTRA(tp)->shape : HPyType_BuiltinShape_Legacy;
 }
 
+static inline cpy_vectorcallfunc _HPyType_get_vectorcall_default(PyTypeObject *tp) {
+    return _is_HPyType(tp) ?
+            _HPyType_EXTRA(tp)->tp_vectorcall_default_trampoline : NULL;
+}
+
+#if HAVE_VECTORCALL_SUPPORT
+static inline void _HPy_set_vectorcall_func(PyTypeObject *tp, PyObject *o, cpy_vectorcallfunc f) {
+    const Py_ssize_t offset = tp->tp_vectorcall_offset;
+    assert(offset > 0);
+    memcpy((char *) o + offset, &f, sizeof(f));
+}
+
+static inline void _HPy_set_vectorcall_default(PyTypeObject *tp, PyObject *o) {
+    if (PyType_HasFeature(tp, _Py_TPFLAGS_HAVE_VECTORCALL)) {
+        cpy_vectorcallfunc vectorcall_default = _HPyType_get_vectorcall_default(tp);
+        _HPy_set_vectorcall_func(tp, o, vectorcall_default);
+    }
+}
```

Comment:
this name is a bit confusing. At a first glance I interpreted it as "set the default vectorcall for this type", but what it actually does is "set the vectorcall for this object to the default for the type", right?
What about `_HPy_set_vectorcall_to_default` or maybe even `_HPy_vectorcall_init` ?

Reply:
changed to `_HPy_vectorcall_init`